warden/client.go

```diff
@@ -0,0 +1,93 @@
+package warden
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/goto/dex/pkg/errors"
+)
+
+type Client struct {
+	BaseURL string
+	Client  *http.Client
+}
+
+func NewClient(baseURL string) *Client {
+	return &Client{
+		BaseURL: baseURL,
+		Client:  &http.Client{},
+	}
+}
+
+func (c *Client) ListUserTeams(ctx context.Context, req TeamListRequest) ([]Team, error) {
+	const (
+		endpoint      = "/api/v1"
+		userPath      = "/users/"
+		teamsEndpoint = "/teams"
+	)
+	url := fmt.Sprintf("%s%s%s%s%s", c.BaseURL, endpoint, userPath, req.Email, teamsEndpoint)
+	fmt.Println("URL: ", url)
+	httpReq, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.Client.Do(httpReq)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode == http.StatusNotFound {
+			return nil, errors.ErrNotFound.WithMsgf("user with email %s not found", req.Email)
```

Comment:
let's not return
```go
errors.ErrNotFound.WithMsgf("user with email %s not found", req.Email)
```

`warden` client should be clean of dependencies.

it should have it's own error.
we could create a new error in warden such as `ErrUserEmailNotFound`